pylearn2/models/mlp.py

```diff
@@ -2988,15 +2994,26 @@ def initialize_transformer(self, rng):
         """
         if self.irange is not None:
             assert self.sparse_init is None
-            self.transformer = conv2d.make_random_conv2D(
-                irange=self.irange,
-                input_space=self.input_space,
-                output_space=self.detector_space,
-                kernel_shape=self.kernel_shape,
-                subsample=self.kernel_stride,
-                border_mode=self.border_mode,
-                rng=rng)
-        elif self.sparse_init is not None:
+
+            if cudnn2d:
+                self.transformer = cudnn2d.make_random_conv2D(
+                    irange=self.irange,
+                    input_space=self.input_space,
+                    output_space=self.detector_space,
+                    kernel_shape=self.kernel_shape,
+                    subsample=self.kernel_stride,
+                    border_mode=self.border_mode,
+                    rng=rng)
+            else:
+                self.transformer = conv2d.make_random_conv2D(
+                    irange=self.irange,
+                    input_space=self.input_space,
+                    output_space=self.detector_space,
+                    kernel_shape=self.kernel_shape,
+                    subsample=self.kernel_stride,
+                    border_mode=self.border_mode,
+                    rng=rng)
```

Comment:
If you detect this automatically, maybe it would be better to simply import `cudnn2d` as `conv2d`, and avoid that duplication here.
This would also ensure that `make_sparse_....` would use the cudnn version as well..